### PR TITLE
#1030 Fix error in makeSkyImage at ra=0

### PR DESCRIPTION
**Pull request by rmjarvis**
@cwwalter reported an error in making sky images with imsim when the image crosses the ra=0 line.

Basically, the ra jumps abruptly from 359.999 to 0.0000, so the finite difference calculation of pixel area gets some crazy large values for these pixels.

It turns out that the GSFitsWCS implementation doesn't suffer from this problem, since it keeps all ra values near the central value, rather than explicitly wrapping to [0,360).  But Astropy has the same feature as the LSST DM WCS, so it worked to duplicate the error and allow for a unit test.

### CODE REVIEW

**Comment by rmjarvis** [comment]
Chris, let me know whether there is a pressing need to get this into 2.1.x as a bugfix release, or if you are ok with letting the slide until the next regular GalSim release (I'd guess sometime later this summer probably).

**Comment by jmeyers314** [comment]
>It turns out that the GSFitsWCS implementation doesn't suffer from this problem, since it keeps all ra values near the central value, rather than explicitly wrapping to [0,360).

What happens if the image contains the pole?

**Comment by rmjarvis** [comment]
> What happens if the image contains the pole?

That will definitely fail.  And the fix wouldn't be particularly simple.  It already makes the approximation when doing the finite difference that cos(dec) is constant over the image.  At the pole, this factor goes to zero, so the areas would be pretty wrong near there.

**Comment by cwwalter** [comment]
> Chris, let me know whether there is a pressing need to get this into 2.1.x as a bugfix release, or if you are ok with letting the slide until the next regular GalSim release (I'd guess sometime later this summer probably).

Sorry; I was traveling yesterday.  I think letting it slide into the summer is OK.  I'm pretty sure there is no overlap for the DC2 footprint (but we should double check) and for the standalone tests I am doing that triggered this I can just not use RA=0.

**Comment by rmjarvis** [comment]
Actually, that's not true (about using constant cos(dec)).  Looks like it does compute the cosine of the declination of each pixel's dec value.  So maybe it's not so terrible at the pole.  I think it's still going to fail right now, but it might not be impossible to make it work there.

**Comment by rmjarvis** [comment]
OK.  I added a test with an image at the south pole, and it works there too now.  I think the actual pole itself is not super accurate, but at least most of the image is reasonable, and even the pole itself is not completely crazy.

File: tests/test_wcs.py [hunk 1]
```diff
@@ -388,6 +388,11 @@ def do_wcs_image(wcs, name, approx=False):
         area = wcs.pixelArea(galsim.PositionD(x,y))
         np.testing.assert_almost_equal(val/(area*sky_level), 1., digits,
                                        "SkyImage at %d,%d is wrong"%(x,y))
+    # Check that all values are near the same value.  In particular if ra crosses 0, then this
```
[the remark this answers is not in the document]
Comment (rmjarvis) [reply tests/test_wcs.py]: It makes about 3 seconds difference, so I switched to != rather than remove it.

**Comment by rmjarvis** [comment]
Thanks Josh.  I took all your suggestions.  Let's plan to merge this tomorrow (Monday) evening unless there are any objections.  I have #1026 ready to go next, based on this branch as a starting point.

**Pull request merged**